add reference docs for LinstorSatelliteConfiguration (#371)
piraeusdatastore/piraeus-operator: docs/reference/linstorsatelliteconfiguration.md

  property will not be set.

```yaml
apiVersion: piraeus.io/v1
kind: LinstorSatelliteConfiguration
metadata:
  name: ipv6-nodes
spec:
  properties:
    - name: PrefNic
      value: "default-ipv6"
    - name: Aux/example-property
      valueFrom:
        nodeFieldRef: metadata.labels['piraeus.io/example']
    - name: AutoplaceTarget
      valueFrom:
        nodeFieldRef: metadata.annotations['piraeus.io/autoplace']
      optional: yes
```

### `.spec.storagePools`

Configures LINSTOR Storage Pools.

Every Storage Pool needs at least a `name`, and a type. Types are specified by setting a (potentially empty) value on
the matching key. Available types are:

* `lvm`: Configures a [LVM Volume Group](https://linbit.com/drbd-user-guide/drbd-guide-9_0-en/#s-lvm-primer) as storage
  pool. Defaults to using the storage pool name as the VG name. Can be overridden setting `volumeGroup`.
* `lvmThin`: Configures a [LVM Thin Pool](https://man7.org/linux/man-pages/man7/lvmthin.7.html) as storage pool.
  Defaults to using the storage pool name as name for the thin pool volume and the storage pool name prefixed by

>   `linstor_` as the VG name. Can be overridden setting `thinPool` and `volumeGroup`.

Reviewer: ```suggestion
  `linstor_` as the VG name. Can be overridden by setting `thinPool` and `volumeGroup`.
```

Optionally, you can configure LINSTOR to automatically create the backing pools. `source.hostDevices` takes a list
of raw block devices, which LINSTOR will prepare as the chosen backing pool.

#### Example

This example configures three LINSTOR Storage Pools on all satellites:
* A LVM Pool named `vg1`. It will be using the VG `vg1`, which needs to exist on the nodes already.

> * A LVM Thin Pool named `vg1-thin`. It will be using the thin pool `vg1/thin`, which also needs to exist on the nodes.

Reviewer: ```suggestion
* A LVM Thin Pool named `vg1-thin`. It will use the thin pool `vg1/thin`, which also needs to exist on the nodes.
```

> * A LVM Pool named `vg2-from-raw-devices`. It will be using the VG `vg2`, which will be created on demand from the raw

Reviewer: ```suggestion
* A LVM Pool named `vg2-from-raw-devices`. It will use the VG `vg2`, which will be created on demand from the raw
```

  tls.crt: LS0tLS1CRUdJT...
  tls.key: LS0tLS1CRUdJT...
---
apiVersion: piraeus.io/v1
kind: LinstorSatelliteConfiguration
metadata:
  name: satellite-tls
spec:
  internalTLS:
    secretName: my-node-tls
```

#### Example

This example sets up automatic creation of the LINSTOR Satellite TLS secrets using a
cert-manager issuer named `piraeus-root`.

```yaml
apiVersion: piraeus.io/v1
kind: LinstorSatelliteConfiguration
metadata:
  name: satellite-tls
spec:
  internalTLS:
    certManager:
      kind: Issuer
      name: piraeus-root
```

### `.spec.patches`

The given patches will be applied to all resources controlled by the operator. The patches are
forwarded to `kustomize` internally, and take the [same format](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/).

The unpatched resources are available in the [subdirectories of the `pkg/resources/satellite` directory](../../pkg/resources/satellite).

#### Warning

No checks are run on the result of user-supplied patches: the resources are applied
as-is. Patching some fundamental aspect, such as removing a specific volume from a
container may lead to a degraded cluster.

#### Example

> This example configures the LINSTOR Satellite to use the "TRACE" log level, creating a very verbose output.

Reviewer: ```suggestion
This example configures the LINSTOR Satellite to use the "TRACE" log level, creating very verbose output.
```
